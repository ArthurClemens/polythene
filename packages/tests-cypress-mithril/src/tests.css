.page__test__component .align-items-center {
  display: flex;
  align-items: center;
}