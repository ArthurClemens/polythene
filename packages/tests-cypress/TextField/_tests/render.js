
const range = n =>
  Array.apply(null, Array(n)).map((_, i) => i);

const COL_COUNT = 10;
const ROW_COUNT = 90;

export default ({ rootPath, h, TextField }) => {
  const Columns = {
    view: ({ attrs: { row } }) =>
      range(COL_COUNT).map(col =>
        h("td", 
          { key: col },
          h(TextField, {
            testId: `${row}-${col}`,
            label: "Text",
            floatingLabel: true,
            defaultValue: `Text ${row}-${col}` 
          })
        )
      )
  };

  const Rows = {
    view: () =>
      range(ROW_COUNT).map(row =>
        h("tr",
          { key: row },
          h(Columns, { row })
        )
      )
  };
      
  let renderCount = 0;

  return {
    path: `${rootPath}/render`,
    name: "Text Field: render",
    component: {
      view: () => {
        renderCount++;
        return h("div",
          {
            "data-renderer-id": h.displayName
          },
          [
            h("p",
              { key: "count" },
              [
                "view called: ",
                h("span", {
                  "data-test-id": "count",
                }, renderCount),
              ]
            ),
            h("table",
              { key: "table" },
              h("tbody", null,
                h(Rows)
              )
            )
          ]
        );
      }
    }
  };
};
