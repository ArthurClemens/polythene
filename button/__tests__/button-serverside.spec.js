/* global describe, it, expect */
import m from 'mithril';
import { ms, tidyHtml, runServerSnapshots } from '../../scripts/snapshots';
import { tests } from './tests.js';
import render from 'mithril-node-render';
import component from 'polythene/button/button';

const renderer = (cmp, opts, children) => render(m(cmp, opts, children));

runServerSnapshots(tests, renderer);

describe('Button component', () => {
  it('no options', () => {
    const html = renderer(component);
    expect(html).toContain('<a class="pe-button pe-button--text');
  });
  it('option content', () => {
    const html = renderer(component, {content: m('span', 'Content')});
    expect(html).toContain('<span>Content</span>');
  });
  it('option id', () => {
    const html = renderer(component, {id: 'id-x'});
    expect(html).toContain('id="id-x"');
  });
  it('option label', () => {
    const html = renderer(component, {label: 'Button label'});
    expect(html).toContain('Button label');
  });
  it('option class', () => {
    const html = renderer(component, {class: 'class-x'});
    expect(html).toContain('class-x');
  });
  it('option tag', () => {
    const html = renderer(component, {tag: 'button'});
    expect(html).toContain('<button ');
    expect(html).toContain('</button>');
  });
  it('option tabindex', () => {
    const html = renderer(component, {tabindex: 3});
    expect(html).toContain('tabindex="3"');
  });
  it('option before', () => {
    const html = renderer(component, {before: m('span', 'Before')});
    expect(html).toContain('<span>Before</span>');
  });
  it('option after', () => {
    const html = renderer(component, {after: m('span', 'After')});
    expect(html).toContain('<span>After</span>');
  });
});
