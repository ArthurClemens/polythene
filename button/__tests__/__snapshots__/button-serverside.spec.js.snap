exports[`test Option: disabled (false) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Not disabled</div></div></a>"`;

exports[`test Option: disabled (true) 1`] = `"<a class=\"pe-button pe-button--text  pe-button--disabled     \" id=\"\" tabindex=\"-1\" disabled><div class=\"pe-button__content\"><div class=\"pe-button__label\">Disabled</div></div></a>"`;

exports[`test Option: formaction 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\" formaction=\"http://polythene.js.org\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Form action</div></div></a>"`;

exports[`test Option: inactivate (2) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Inactivated for 2s</div></div></a>"`;

exports[`test Option: inactive (false) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Not inactive</div></div></a>"`;

exports[`test Option: inactive (true) 1`] = `"<a class=\"pe-button pe-button--text   pe-button--inactive    \" id=\"\" tabindex=\"-1\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Inactive</div></div></a>"`;

exports[`test Option: ink (false) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">No ink</div></div></a>"`;

exports[`test Option: label 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Label</div></div></a>"`;

exports[`test Option: selected 1`] = `"<a class=\"pe-button pe-button--text pe-button--selected      \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Selected</div></div></a>"`;

exports[`test Option: url (with oncreate) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\" href=\"/shadow\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Go to /#/shadow</div></div></a>"`;

exports[`test Option: url (without oncreate) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\" href=\"/shadow\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__wash\"></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">Go to /shadow</div></div></a>"`;

exports[`test Option: wash (false) 1`] = `"<a class=\"pe-button pe-button--text       \" id=\"\" tabindex=\"0\"><div class=\"pe-button__content\"><div class=\"pe-ripple pe-ripple--constrained \" id=\"\"><div class=\"pe-ripple__mask\"><div class=\"pe-ripple__waves\"></div></div></div><div class=\"pe-button__focus\"></div><div class=\"pe-button__label\">No wash</div></div></a>"`;
